staging/src/k8s.io/client-go/rest/watch/encoder_test.go

```diff
@@ -24,49 +24,48 @@ import (
 	apiequality "k8s.io/apimachinery/pkg/api/equality"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	runtimejson "k8s.io/apimachinery/pkg/runtime/serializer/json"
 	"k8s.io/apimachinery/pkg/runtime/serializer/streaming"
 	"k8s.io/apimachinery/pkg/watch"
-	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/pkg/api/v1"
 	restclientwatch "k8s.io/client-go/rest/watch"
-
-	_ "k8s.io/client-go/pkg/api/install"
 )
 
 func TestEncodeDecodeRoundTrip(t *testing.T) {
 	testCases := []struct {
 		Type   watch.EventType
 		Object runtime.Object
-		Codec  runtime.Codec
 	}{
 		{
 			watch.Added,
-			&api.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
-			api.Codecs.LegacyCodec(v1.SchemeGroupVersion),
+			&v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
 		},
 		{
 			watch.Modified,
-			&api.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
-			api.Codecs.LegacyCodec(v1.SchemeGroupVersion),
+			&v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
 		},
 		{
 			watch.Deleted,
-			&api.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
-			api.Codecs.LegacyCodec(v1.SchemeGroupVersion),
+			&v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
 		},
 	}
 	for i, testCase := range testCases {
 		buf := &bytes.Buffer{}
 
-		codec := testCase.Codec
-		encoder := restclientwatch.NewEncoder(streaming.NewEncoder(buf, codec), codec)
+		jsonSerializer := runtimejson.NewSerializer(runtimejson.DefaultMetaFactory, scheme.Scheme, scheme.Scheme, false)
```

Comment:
It would be better to make a function than copy-pasting these three lines in multiple places :)